showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseBrowserPropertyWriter.kt

```diff
@@ -0,0 +1,214 @@
+package com.airbnb.android.showkase.processor.writer
+
+import androidx.room.compiler.processing.XFiler
+import androidx.room.compiler.processing.XProcessingEnv
+import androidx.room.compiler.processing.addOriginatingElement
+import androidx.room.compiler.processing.writeTo
+import com.airbnb.android.showkase.processor.models.ShowkaseMetadata
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.asTypeName
+
+class ShowkaseBrowserPropertyWriter(private val environment: XProcessingEnv) {
+    internal fun generateMetadataPropertyFile(
+        showkaseComponentMetadata: Set<ShowkaseMetadata.Component>,
+        showkaseColorMetadata: Set<ShowkaseMetadata>,
+        showkaseTypographyMetadata: Set<ShowkaseMetadata>,
+        rootModulePackageName: String,
+    ): Triple<Pair<List<String>, List<String>>, List<String>, List<String>> {
```

Comment:
this return type is pretty opaque. can make a data class to wrap the data cleanly instead